src/controllers/authentication.ts

```diff
@@ -0,0 +1,53 @@
+import jwt, { JwtPayload } from 'jsonwebtoken'
+import express from 'express'
+import User from '../database/models/userModel'
+import Token from '../database/models/tokenModel'
+
+export const validateAndRetrieveToken = async (token: string) => {
+  try {
+    const decodedToken = jwt.verify(token, process.env.JWT_SECRET, {
+      ignoreExpiration: false,
+    }) as JwtPayload
+    const myToken = await Token.findOne({ where: { token } })
```

Comment:
I think this function can be used as helpers not a controller, try to move it into helpers folder.

Reply:
The reason it is here is because it is specifically for this verify token I separated them because of the codeclimate issue of cognitive comprexity and for easy testing